kmmsdk/src/commonMain/kotlin/com/lokate/kmmsdk/LokateSDK.kt

```diff
@@ -1,103 +1,227 @@
+                        activeBeacons.remove(it)
                     }
                 }
             }
+        }
+    }
 
+    private fun flowEmitter() {
         lokateScope.launch {
-            beaconFlowHandler(newComer, alreadyIn, gone)
+            newComer.collect {
+                log.d { "Newcomer: $it" }
+                beaconRepository.sendBeaconEvent(
+                    EventRequest(
+                        "umut",
+                        it.beaconUUID,
+                        it.major.toString(),
+                        it.minor.toString(),
+                        EventStatus.ENTER,
+                        it.seen,
+                    ),
+                ).also { log.d { "Send newcomer in: $it" } }
+            }
+        }
+        lokateScope.launch {
+            alreadyIn.collect {
+                log.d { "Already in: $it" }
+                beaconRepository.sendBeaconEvent(
+                    EventRequest(
+                        "umut",
+                        it.beaconUUID,
+                        it.major.toString(),
+                        it.minor.toString(),
+                        EventStatus.STAY,
+                        it.seen,
+                    ),
+                ).also { log.d { "Send already in: $it" } }
+            }
+        }
+        lokateScope.launch {
+            gone.collect {
+                log.d { "Gone: $it" }
+                beaconRepository.sendBeaconEvent(
+                    EventRequest(
+                        "umut",
+                        it.beaconUUID,
+                        it.major.toString(),
+                        it.minor.toString(),
+                        EventStatus.EXIT,
+                        it.seen,
+                    ),
+                ).also { log.d { "Send gone: $it" } }
+            }
         }
-        // beaconScanner.start()
-        // results = beaconScanner.observeResults()
     }
 
-    private suspend fun beaconFlowHandler(
-        beaconScannerFlow: Flow<BeaconScanResult>,
-        alreadyIn: Flow<BeaconScanResult?>,
-        gone: Flow<BeaconScanResult>,
-    ) {
-        // first look for the beacons that are already in the list
-        alreadyIn.collect {
-            it?.let {
-                activeBeacons.removeAll {
+    private fun scanResultHandler(beaconScannerFlow: Flow<BeaconScanResult>) {
+        lokateScope.launch {
+            beaconScannerFlow.collect { scan ->
+                activeBeacons.firstOrNull {
                     it.beaconUUID == it.beaconUUID &&
-                        it.major == it.major && it.minor == it.minor
+                        it.major == it.major &&
+                        it.minor == it.minor
                 }
-                activeBeacons.add(it)
-                // send stay event
-                Napier.d("Beacon is still here $it")
+                    ?.let {
+                        alreadyIn.emit(it.copy(seen = getTimeMillis()))
```

Comment:
Why do we update the "seen" here? As far as I understand, beaconScannerFlow() contains always updated values since toBeaconScanResult() function calculates current time just before emitting BeaconScanResult to beaconScannerFlow. Am I missing something?

Reply:
just alreadyIn.emit(scan) might be sufficient

Reply:
just alreadyIn.emit(scan) might be sufficient

Reply:
One more thing; checkGone() function checks the timestamps of BeaconScanResults inside activeBeacons() set, but in this code block, "seen" field of the respective BeaconScanResult inside activeBeacons() is not updated.